### DISCUSSION TITLE
Freezed class as part of a library

### DISCUSSION THREAD

**Discussion by lazarohcm** [rrousselGit/freezed#727] [Q&A]
Is it possible to have a class that uses freezed, which is part of a library? 
I've tried it, but couldn't figure out if I was declaring it wrong or not or if it is just not possible due to the restrictions of `part` and `part of` usage; 

Example bellow

_news.dart_

``` dart
library news;
import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:flutter/foundation.dart';

part 'news_model.dart';
part 'news_model.freezed.dart;
```

_news_model.dart_

``` dart
@freezed
class NewsModel with _$NewsModel {
  const NewsModel._();

  @JsonSerializable()
  factory NewsModel(
    String? id,
  ) = _NewsModel;
}
```

**Comment by SunlightBro** [comment]
Checkout darts documentation about [library](https://dart.dev/guides/libraries/create-library-packages#organizing-a-library-package)

> **Note**: You may have heard of the `part` directive, which allows you to split a library into multiple Dart files. We recommend that you avoid using `part` and create mini libraries instead.

So for your example, you should do the following:

`lib/news.dart`
```dart
library news;
export 'src/news_model.dart' show NewsModel;

```
`lib/src/news_model.dart`
```dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:flutter/foundation.dart';

part 'news_model.freezed.dart;

@freezed
class NewsModel with _$NewsModel {...}
```

**Reply by lazarohcm** [reply]
Thanks, @SunlightBro!

I've it implemented like that already, was just exploring other design options.
